src/components/Table/index.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+import { composeTableHead } from '../TableHead';
+import './styles.css';
+
+export enum SortDirection {
+  asc = 'ASC',
+  desc = 'DESC',
+}
+
+export type ColumnType<T extends string | number> = {
```

Comment:
I would probably add 
```ts
type EnumValue = string | number;
```
And replace all the `string | number` with is. Not necessary, but it makes it more clear what `string | number` means in this context